src/hotspot/share/runtime/thread.cpp

```diff
@@ -145,6 +145,14 @@ Thread::Thread(MemTag mem_tag) {
   MACOS_AARCH64_ONLY(DEBUG_ONLY(_wx_init = false));
 }
 
+address Thread::stack_base() const {
+  // Note: can't report Thread::name() here as that can require a ResourceMark which we
+  // can't use because this gets called too early in the thread initialization.
+  assert(_stack_base != nullptr, "Stack base not yet set for thread id:%d (0 if not set)",
+         osthread() != nullptr ? osthread()->thread_id() : 0);
+  return _stack_base;
+}
+
```

Comment:
Can't we still have this as inlined version? Either move this to thread.inline.hpp or give us a release variant without assert in Thread body in thread.hpp.

Reply:
Yeah. I think we can "just" outline the checking method like:

```
thread.hpp:
  address stack_base() const           { assert_stack_base(); return _stack_base; }
  void assert_stack_base() const NOT_DEBUG_RETURN;

thread.cpp:
#ifdef ASSERT
void  Thread::assert_stack_base() const {
  // Note: can't report Thread::name() here as that can require a ResourceMark which we
  // can't use because this gets called too early in the thread initialization.
  assert(_stack_base != nullptr, "Stack base not yet set for thread id:%d (0 if not set)",
         osthread() != nullptr ? osthread()->thread_id() : 0);
}
#endif
```